pkg/platform/platform_versioner.go

```diff
@@ -0,0 +1,121 @@
+package platform
+
+import (
+	"encoding/json"
+	openapi_v2 "github.com/googleapis/gnostic/OpenAPIv2"
```

Comment:
Is this in lieu of `github.com/go-openapi/spec`, which is already imported? Is that not backward compatible, or otherwise not assignable?

Reply:
I've spent a while on it now, but I've been unable to find a way to map out of Gnostic's generated model used by K8S ([Discovery](https://github.com/kubernetes/client-go/tree/master/discovery)) into the openapi spec. The interface I've added (Discoverer) is just an abstraction to allow for test injection of a Discovery mock, so neither have I found a way to alter the declared methods without breaking away from the correct interface needed for injection. Any thoughts?

Reply:
Ok, I don't know for a fact that it can be done, so I just wanted you to investigate. Never mind, then, thanks!